mkosi/__init__.py

```diff
@@ -4277,6 +4305,29 @@ def parse_args_file_group(argv_post_parsed: List[str], default_path: str) -> Com
     return parser.parse_args(argv_post_parsed, CommandLineArguments())
 
 
+def parse_args_distribution_group(argv_post_parsed: List[str], default_path: str, distribution: str) -> CommandLineArguments:
+    all_defaults_files = []
+    defaults_dir = default_path + ".d"
+
+    distribution_dir = os.path.join(defaults_dir, distribution)
+    if os.path.isdir(distribution_dir):
+        for distribution_file in sorted(os.listdir(distribution_dir)):
+            distribution_path = os.path.join(distribution_dir, distribution_file)
+            if os.path.isfile(distribution_path):
+                all_defaults_files.append(ArgumentParserMkosi.fromfile_prefix_chars + distribution_path)
+
+    # Insert the distro specific config files after the rest of the config files so they override these.
+    for i, v in enumerate(argv_post_parsed):
+        if not v.startswith(ArgumentParserMkosi.fromfile_prefix_chars):
+            argv_post_parsed[i:i] = all_defaults_files
+            break
+        # Append to the end if the args only contain files and no regular args.
+        if i == len(argv_post_parsed) - 1:
+            argv_post_parsed += all_defaults_files
```

Comment:
Although uncommon, this is *the* textbook case for `else` on the for loop
```suggestion
        # Append to the end if the args only contain files and no regular args.
    else:
        argv_post_parsed += all_defaults_files
```

Reply:
Pretty nifty!